misc/rapiddisk-rootdev/ubuntu/rapiddisk

```diff
@@ -14,12 +14,13 @@ esac
 
 # Begin real processing below this line
 
-modprobe -v rapiddisk
-modprobe -v rapiddisk-cache
+if [ -x /usr/sbin/rapiddisk_KERNELVERSION ] ; then
```

Comment:
I am confused here. Where does /usr/sbin/rapiddisk_KERNELVERSION come from? And when does that exact binary with string name get installed because below in ubuntu/rapiddisk_hook, it does so in /sbin and not /usr/sbin. Also, we are we creating multiple copies of the same binary?

Reply:
> I am confused here. Where does /usr/sbin/rapiddisk_KERNELVERSION come from? Is the rapiddisk executable added to the initrd file with a custom name?

Yes.

> And when does that exact binary with string name get installed because below in ubuntu/rapiddisk_hook, it does so in /sbin and not /usr/sbin

This is an initramfs behaviour. `/sbin` is converted to `/usr/sbin` internally. So, when, using its macros `/sbin` is to be used, but in the final initrd file the execs will be installed in `/usr/sbin`

> Also, we are we creating multiple copies of the same binary?

No, in each initrt file there is only one copy of the binary, included with a modified name. This is what happens under Ubuntu. On CentOS things are a bit different.

Reply:
OK. I understand. Thank you for taking the time to explain this.